OpenSim/Simulation/Model/KneeLigament.cpp

```diff
@@ -0,0 +1,300 @@
+ * For this object, that entails calculating and storing the
+ * length in the current body position.
+ *
+ * @param aScaleSet XYZ scale factors for the bodies.
+ */
+void KneeLigament::preScale(const SimTK::State& s, const ScaleSet& aScaleSet)
+{
+    updGeometryPath().preScale(s, aScaleSet);
+}
+
+//_____________________________________________________________________________
+/**
+ * Scale the ligament.
+ *
+ * @param aScaleSet XYZ scale factors for the bodies
+ * @return Whether or not the ligament was scaled successfully
+ */
+void KneeLigament::scale(const SimTK::State& s, const ScaleSet& aScaleSet)
+{
+    updGeometryPath().scale(s, aScaleSet);
+}
+
+//_____________________________________________________________________________
+/**
+ * Perform computations that need to happen after the ligament is scaled.
+ * For this object, that entails comparing the length before and after scaling,
+ * and scaling the resting length a proportional amount.
+ *
+ * @param aScaleSet XYZ scale factors for the bodies.
+ */
+void KneeLigament::postScale(const SimTK::State& s, const ScaleSet& aScaleSet)
+{
+    GeometryPath& path          = updGeometryPath();
+    double&       restingLength = upd_resting_length();
+
+    path.postScale(s, aScaleSet);
+
+    if (path.getPreScaleLength(s) > 0.0)
+    {
+        double scaleFactor = path.getLength(s) / path.getPreScaleLength(s);
+
+        // Scale resting length by the same amount as the change in
+        // total ligament length (in the current body position).
+        restingLength *= scaleFactor;
+
+        path.setPreScaleLength(s, 0.0);
+    }
+}
+
+const double& KneeLigament::getTension(const SimTK::State& s) const
+{
+    return getCacheVariableValue<double>(s, "tension"); 
+}
+
+
+//=============================================================================
+// COMPUTATION
+//=============================================================================
+/**
+ * Compute the moment-arm of this muscle about a coordinate.
+ */
+double KneeLigament::computeMomentArm(const SimTK::State& s, Coordinate& aCoord) const
+{
+    return getGeometryPath().computeMomentArm(s, aCoord);
+}
+
+
+
+void KneeLigament::computeForce(const SimTK::State& s, 
+                              SimTK::Vector_<SimTK::SpatialVec>& bodyForces, 
+                              SimTK::Vector& generalizedForces) const
+{
+    const GeometryPath& path = getGeometryPath();
+    const double& restingLength = get_resting_length();
+
+    double force = 0;
+
+    if (path.getLength(s) <= restingLength){
+        setCacheVariableValue<double>(s, "tension", force);
+        return;
+    }
+    
+    // evaluate normalized tendon force length curve
+	force = computeLigamentForceStrain(path.getLength(s));
+    setCacheVariableValue<double>(s, "tension", force);
+
+    OpenSim::Array<PointForceDirection*> PFDs;
+    path.getPointForceDirections(s, &PFDs);
+
+    for (int i=0; i < PFDs.getSize(); i++) {
```

Comment:
The following code for applying point forces can be replaced by:
`path.addInEquivalentForces(s, force, bodyForces, generalizedForces);`